server/app/mutations/grid_services/update.rb

```diff
@@ -120,7 +120,7 @@ def build_grid_service_envs(env)
       new_env = GridService.new(env: env).env_hash
       current_env = self.grid_service.env_hash
       new_env.each do |k, v|
-        if v.empty? && current_env[k]
+        if (v.nil? || v.empty?) && current_env[k]
```

Comment:
It would be better to validate and reject invalid `env` strings earlier, rather than fixing the `execute` -> `build_grid_service_envs` to deal with the result of trying to parse invalid envs.... fix the [`GridServices::Common` validation for the `env` array](https://github.com/kontena/kontena/blob/v1.2.1/server/app/mutations/grid_services/common.rb#L175) to use something like `string matches: /.*=.*/`?

However, that would result in the stacks mutations outcome error handling stumbling over the resulting `Mutations::ErrorArray` as per https://github.com/kontena/kontena/issues/2238#issuecomment-298569759, but fixed in #2058.